openapi/data_repository_service.swagger.yaml

```diff
@@ -420,35 +347,26 @@ definitions:
       name:
         type: string
         description: >-
-          A name declared by the Bundle author that must be
-          used when materialising the associated data object, 
+          A name declared by the bundle author that must be
+          used when materialising this object, 
           overriding any name directly associated with the object itself.
           This string MUST NOT contain any slashes.
       id:
         type: string
         description: >-
-          A DRS identifier of a Data Object or a nested Data Bundle.
+          A DRS identifier of an `Object` (either a single blob or a nested bundle).
       drs_uri:
         type: array
         description: >-
           A list of full DRS identifier URI paths
-          that may be used obtain the Data Object or Data Bundle.
+          that may be used to obtain the object.
           These URIs may be external to this DRS instance.
         example:
           drs://example.com/ga4gh/drs/v1/objects/{object_id}
         items:
           type: string
-      type:
```

Comment:
Why did this type information get removed? How are we to know what kind of object it is?

Reply:
agree, should keep type

Reply:
I don't feel strongly here. I removed the type because callers can now just call `/objects/{object_id}` to get info about it including its type, which they need to do in all cases. So it felt redundant and like unneeded work for server developers to populate here. Given that, what do the two you think?

Reply:
I would prefer a restructuring so that the response was:
`[id, name, type (bundle or blob), bundle: BundleObject, object: BlobObject]`
and separate the two object structures. We can certainly share substructures of them, but not within this response.

That makes it obvious to the client which element to attend to based on the type. In code, it gives you one reference(pointer) to traverse. It also makes it clear how to extend the interface with additional types.

Reply:
Dan, are you suggesting that we embed the full object in the `contents` array, or am I misunderstanding? (We can talk on the WS call in a couple of minutes if easier.)

Reply:
Had a closer look, didn't realize this `type` is under `BundleObject `. I thought `BundleObject` and *blob* object are merged, ie, there is no `BundleObject` any more. If that's the intention, I think it makes sense to keep `type` for the new unified `object`.

Reply:
I agree, I think this will be super useful when we have other object types in the future... like BigTable entries for example

Reply:
Notes from the WS call:
- Dan agrees we don't need to include the `type` in the `contents` array, but it could be useful -- he could take it or leave it
- Dan would prefer to restructure the entire response, to have a type at the top, and sub-structures below. But is okay either way.
- David pointed out that there the two sub-types are very similar (for blobs `access-methods` is required and `contents` isn't valid; for bundles `access-methods` is optional and `contents` is required)
- we want to rename `BundleObject` to `ContentsObject`